WordPress/src/main/java/org/wordpress/android/ui/posts/PostsListActivity.java

```diff
@@ -89,10 +90,16 @@ private void handleIntent(Intent intent) {
             actionBar.setDisplayHomeAsUpEnabled(true);
         }
 
+        PostModel targetPost = null;
+        int targetPostId = intent.getIntExtra(EXTRA_TARGET_POST_ID, 0);
+        if (targetPostId > 0) {
+            targetPost = mPostStore.getPostByLocalPostId(intent.getIntExtra(EXTRA_TARGET_POST_ID, 0));
+        }
+
         mPostList = (PostsListFragment) getFragmentManager().findFragmentByTag(PostsListFragment.TAG);
-        if (mPostList == null || siteHasChanged || pageHasChanged) {
+        if (mPostList == null || siteHasChanged || pageHasChanged || targetPost != null) {
             PostsListFragment oldFragment = mPostList;
-            mPostList = PostsListFragment.newInstance(mSite, mIsPage);
+            mPostList = PostsListFragment.newInstance(mSite, mIsPage, targetPost);
```

Comment:
> In this case we're passing the PostModel directly to a method, it's not being parceled, so #5856 shouldn't apply.

Forgive me, I wasn't clear:  I was trying to acknowledge how the same treatment to this case is given within the fragment by using the ID instead of the PostModel. I should have said something like "it's great to see we're using this in the same way here".